File: serverless/Node/lambda_listar.js
// routes/archivos.js

const express = require("express");
const jwt = require("jsonwebtoken");
const sql = require("mssql");

const router = express.Router();

const SECRET_KEY = process.env.SECRET_KEY;

// 🔗 Configuración SQL Server
const dbConfig = {
    user: process.env.DB_USER,
    password: process.env.DB_PASSWORD,
    server: process.env.DB_HOST,
    database: process.env.DB_NAME,
    options: {
        encrypt: false, // true si usas Azure
        trustServerCertificate: true,
    },
};

// 📂 GET /archivos
router.get("/archivos", async (req, res) => {
    const authHeader = req.headers["authorization"];

    // 🔐 Validar token
    if (!authHeader) {
        return res.status(401).json({ detail: "Token requerido" });
    }

    let user_id;

    try {
        const token = authHeader.split(" ")[1];
        const payload = jwt.verify(token, SECRET_KEY);
        user_id = payload.user_id;
    } catch (error) {
        return res.status(401).json({ detail: "Token inválido" });
    }

    try {
        // 🗄️ Conectar a BD
        const pool = await sql.connect(dbConfig);

        const result = await pool
            .request()
            .input("user_id", sql.Int, user_id)
            .query(`
                SELECT file_name, file_type, file_url
                FROM archivos
                WHERE user_id = @user_id
            `);

        const archivos = result.recordset.map((row) => {
            let tipo = String(row.file_type).toLowerCase();
            let tipo_normalizado;

            // 🔥 NORMALIZACIÓN (igual que Python)
            if (["jpg", "jpeg", "png", "gif", "webp"].includes(tipo)) {
                tipo_normalizado = "Imagen";
            } else if (["pdf", "doc", "docx", "xls", "xlsx", "txt"].includes(tipo)) {
                tipo_normalizado = "Documento";
            } else if (["imagen", "image"].includes(tipo)) {
                tipo_normalizado = "Imagen";
            } else if (["documento", "document"].includes(tipo)) {
                tipo_normalizado = "Documento";
            } else {
                tipo_normalizado = "Documento";
            }

            return {
                file_name: row.file_name,
                file_type: tipo_normalizado,
                file_url: row.file_url,
            };
        });

        return res.json(archivos);

    } catch (error) {
        console.error("Error al listar archivos:", error);
        return res.status(500).json({ detail: "Error interno del servidor" });
    }
});

module.exports = router;

File: serverless/Python/lambda_listar.py
from fastapi import APIRouter, Header, HTTPException
import jwt
import os
import pymssql

router = APIRouter()

SECRET_KEY = os.getenv("SECRET_KEY")

def get_db_connection():
    return pymssql.connect(
        host=os.environ["DB_HOST"],
        user=os.environ["DB_USER"],
        password=os.environ["DB_PASSWORD"],
        database=os.environ["DB_NAME"]
    )

@router.get("/archivos")
def listar_archivos(authorization: str = Header(None)):
    #  Validar token
    if not authorization:
        raise HTTPException(status_code=401, detail="Token requerido")

    try:
        token = authorization.split(" ")[1]
        payload = jwt.decode(token, SECRET_KEY, algorithms=["HS256"])
        user_id = payload["user_id"]
    except Exception:
        raise HTTPException(status_code=401, detail="Token inválido")

    # Conexión BD
    conn = get_db_connection()
    cursor = conn.cursor()

    cursor.execute("""
        SELECT file_name, file_type, file_url
        FROM archivos
        WHERE user_id = %s
    """, (user_id,))

    rows = cursor.fetchall()

    archivos = []

    for row in rows:
        file_name = row[0]
        file_type_db = row[1]
        file_url = row[2]

        # NORMALIZACIÓN DEL TIPO (CLAVE PARA LOS FILTROS)
        tipo = str(file_type_db).lower()

        if tipo in ["jpg", "jpeg", "png", "gif", "webp"]:
            tipo_normalizado = "Imagen"
        elif tipo in ["pdf", "doc", "docx", "xls", "xlsx", "txt"]:
            tipo_normalizado = "Documento"
        elif tipo in ["imagen", "image"]:
            tipo_normalizado = "Imagen"
        elif tipo in ["documento", "document"]:
            tipo_normalizado = "Documento"
        else:
            # fallback por seguridad
            tipo_normalizado = "Documento"

        archivos.append({
            "file_name": file_name,
            "file_type": tipo_normalizado,
            "file_url": file_url
        })

    cursor.close()
    conn.close()

    return archivos